scaleway/resource_secret_version.go

```diff
@@ -0,0 +1,189 @@
+package scaleway
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	secret "github.com/scaleway/scaleway-sdk-go/api/secret/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewaySecretVersion() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalewaySecretVersionCreate,
+		ReadContext:   resourceScalewaySecretVersionRead,
+		UpdateContext: resourceScalewaySecretVersionUpdate,
+		DeleteContext: resourceScalewaySecretVersionDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Default: schema.DefaultTimeout(defaultSecretTimeout),
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"secret_id": {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "The secret ID associated with this version",
+				DiffSuppressFunc: diffSuppressFuncLocality,
+			},
+			"data": {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "The data payload of your secret version on 64 mode",
+				Sensitive:        true,
+				ForceNew:         true,
+				DiffSuppressFunc: diffSuppressFuncBase64,
+				StateFunc: func(i interface{}) string {
+					return base64Encoded([]byte(i.(string)))
+				},
+			},
```

Comment:
The current issue with getting a base64 encoded value from terraform is that SDK will base64 the given data anyway, leading to a double base64 encoded data if we do not decode it before passing to SDK.
This make use requesting a base64 encoded data to decode it then encode it again...

Reply:
I've just removed the base64 restriction on the create  and only kept it on the state